addons/calendar/models/calendar_event.py

```diff
@@ -547,31 +547,38 @@ def create(self, vals_list):
         all_models = self.env['ir.model'].sudo().browse(model_ids)
         valid_models = all_models.filtered(lambda m: m.is_mail_activity)
         if meeting_activity_types and not defaults.get('activity_ids'):
+            activity_ids_groups = {}
```

Comment:
if you use a defaultdict you can simplify the code below.
```py
activity_ids_groups = defaultdict(type_you_need)
```
I would prefer a name like grouped_activity_values or something because these are not ids.